tests/test_app_callbacks.py

```diff
@@ -0,0 +1,61 @@
+data = {'processed_df': '{"columns":["Species","Subspecies","View","Sex","hybrid_stat","lat","lon","lat-lon","Samples_at_locality","Species_at_locality","Subspecies_at_locality"],"index":[0,1,2,3,4,5,6,7,8,9],"data":[["erato","notabilis","unknown","unknown","subspecies synonym",-1.583333333,-77.75,"-1.583333333|-77.75",1,"erato","notabilis"],["erato","petiverana","ventral","male","valid subspecies",18.66666667,-96.98333333,"18.66666667|-96.98333333",1,"erato","petiverana"],["unknown","petiverana","ventral","male","valid subspecies","unknown",-84.68333333,"unknown|-84.68333333",1,"unknown","petiverana"],["erato","phyllis","dorsal","male","subspecies synonym",-27.45,-58.98333333,"-27.45|-58.98333333",1,"erato","phyllis"],["unknown","plesseni","ventral","male","valid subspecies",-1.4,"unknown","-1.4|unknown",1,"unknown","plesseni"],["melpomene","unknown","ventral","male","subspecies synonym",-13.36666667,-70.95,"-13.36666667|-70.95",1,"melpomene","unknown"],["melpomene","rosina_S","dorsal","male","valid subspecies",9.883333333,-83.63333333,"9.883333333|-83.63333333",1,"melpomene","rosina_S"],["erato","guarica","dorsal","female","valid subspecies",4.35,-74.36666667,"4.35|-74.36666667",1,"erato","guarica"],["melpomene","plesseni","ventral","male","subspecies synonym",-1.583333333,"unknown","-1.583333333|unknown",1,"melpomene","plesseni"],["melpomene","nanna","unknown","male","valid subspecies",-20.33333333,-40.28333333,"-20.33333333|-40.28333333",1,"melpomene","nanna"]]}',
+        'all_species': {'Erato': ['Any-Erato', 'notabilis', 'petiverana', 'phyllis', 'guarica'], 'Unknown': ['Any-Unknown', 'petiverana', 'plesseni'], 'Melpomene': ['Any-Melpomene', 'unknown', 'rosina_S', 'plesseni', 'nanna'], 'Any': ['Any', 'notabilis', 'petiverana', 'phyllis', 'plesseni', 'unknown', 'rosina_S', 'guarica', 'nanna']}, 
+        'mapping': True, 
+        'images': True}
+jsonified_data = json.dumps(data)
+
+
+def test_update_dist_view_call():
+    # Test hist div callback (no clicks or button)
+    output = update_dist_view(0, [], jsonified_data)
+    j_output = json.dumps(output, cls = plotly.utils.PlotlyJSONEncoder)
+    assert "Show Map View" in j_output
+    
+    # Test map div callback (1 click and "Show Map View")
+    output2 = update_dist_view(1, ["Show Map View"], jsonified_data)
+    j_output2 = json.dumps(output2, cls = plotly.utils.PlotlyJSONEncoder)
+    assert "Show Histogram" in j_output2
+
+    # Test hist div callback (clicks and "Show Histogram")
+    output3 = update_dist_view(2, children = "Show Histogram", jsonified_data = jsonified_data)
+    j_output3 = json.dumps(output3, cls = plotly.utils.PlotlyJSONEncoder)
+    assert "Show Map View" in j_output3
+
+
+def test_update_dist_plot_call():
+    # Check for proper type of fig (Histplot output)
+    output = update_dist_plot('Species', 'View', 'alpha', "Show Map View", jsonified_data)
+    assert str(type(output['data', 0])) == "<class 'plotly.graph_objs._histogram.Histogram'>"
```

Comment:
I wouldn't check for the type being hidden class. With python there is a convention that things that start with "_" are private and not considered part of the package API. So this could change without warning in future releases and break the test. There is a similar issue with update_dist_plot below. Perhaps we could just remove the asserts. Perhaps just having code that exercises the function (so we know if it starts blowing up) is enough in this case.

Reply:
Would it work to do:
`output['data', 0].type`, which produces 'scattergeo' (the dictionary value associated with `key` 'type)?

Reply:
That sounds like a more resilient way to check the result.